resources/append_if_no_line.rb

```diff
@@ -21,3 +20,18 @@
     not_if { ::File.exist?(new_resource.path) && !current.grep(regex).empty? }
   end
 end
+
+action_class do
+  def regex
+    string = Regexp.escape(new_resource.line)
```

Comment:
I'd probably write the logic in this method as

leading = new_resource.ignore_leading_whitespace ? ''\s*' : ''
trailing = new_resource.ignore_trailing_whitespace ? ''\s*' : ''
/^#{leading}#{string}#{trailing}$/

Reply:
Yep, I had similar thoughts. I'll update.